src/Compilers/CSharp/Portable/BoundTree/BoundObjectCreationExpression.cs

```diff
@@ -5,14 +5,18 @@
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
-    internal partial class BoundObjectCreationExpression
+    internal partial class BoundObjectCreationExpression : IBoundAnonymousObjectCreation
```

Comment:
>IBoundAnonymousObjectCreation [](start = 59, length = 29)

The requirement to implement ```IBoundAnonymousObjectCreation``` doesn't feel right. I believe ```BoundObjectCreationExpression``` never represents an anonymous object creation. If we need to store ```BoundAnonymousObjectCreationExpression``` and ```BoundObjectCreationExpression``` in the same field or parameter we can simply use BoundExpression (constructors can be strongly typed) and then extract data from them by checking the kind.